simulation-scripts/perturb.sh

```diff
@@ -51,68 +38,37 @@ THIS_SCRIPT="${DIR}/$(basename "$0")"
 
 # required defaults
 declare -a ARGUMENTS
-EXPECTED_NUM_ARGUMENTS=1
 ARGUMENTS=()
 SCENARIO=''
 MASTER_HOST=""
 SLAVE_HOSTS=""
 BASTION_HOST=""
-IS_TEST_ONLY=0
-IS_FORCE=0
 
 main() {
 
-  [[ $# = 0 && "${EXPECTED_NUM_ARGUMENTS}" -gt 0 ]] && usage
-
   parse_arguments "$@"
-  validate_arguments "$@"
 
-  local SCENARIO_DIR="scenario/${SCENARIO}/"
+  [[ "${#ARGUMENTS[@]}" -gt 1 ]] && error "Only one scenario accepted"
 
-  if [[ "${IS_TEST_ONLY:-}" == 1 ]]; then
-    success "In test mode, skipping deployment of ${SCENARIO}"
-  else
+  [[ ${MASTER_HOST:-} ]] || error "--master must be an IP or hostname, or comma-delimited list"
+  [[ ${SLAVE_HOSTS:-} ]] || error "--slaves must be an IP or hostname, or comma-delimited list"
 
-    source test-func.sh
+  SCENARIO="${ARGUMENTS[0]:-}" || true
 
-    local FOUND_SCENARIO
-    if FOUND_SCENARIO=$(find_scenario); then
-      if [[ "${IS_FORCE}" != 1 && "${FOUND_SCENARIO}" == "${SCENARIO}" ]]; then
-        if ! is_special_scenario; then
-          error "Scenario ${SCENARIO} already deployed, reset deployment with 'cleanup' first"
-        fi
-      fi
-      info "Found scenario ${FOUND_SCENARIO}"
-    fi
+  local SCENARIO_DIR="scenario/${SCENARIO}/"
 
-    info "Running ${SCENARIO_DIR} against ${MASTER_HOST}"
+  if [ ! -d "${SCENARIO_DIR}" ]; then
+    error "${SCENARIO_DIR} not found"
```

Comment:
This error checking is only validating local file existence. Removed code prevents the same scenario being deployed twice as these are not guaranteed to be idempotent -- WDYT?

Reply:
Potentially, but there is going to be another PR which ensures if any scenario (not just a potential duplicate deployment) has been deloyed and errors if so, which I think will cover this situation.